### PR TITLE
Single Record Throughput Performance Tweaks

### PR DESCRIPTION
**Pull request by olirice**
## What kind of change does this PR introduce?
- Removes dead code
- Inline single use functions to reduce function call overhead
- Combines multiple data structures into `cdc.wal_column`
- Replaces `information_schema` based permissions check with `pg_catalog.has_column_privilege` (big impact)
- Adds a helper utility for benchmarking (during tests only)

## Additional context
Overall impact on processing time:
- decreases processing time from ~11ms to ~5ms for a single record with a single subscriber

### CODE REVIEW

File: sql/walrus--0.1.sql [hunk 1]
```diff
@@ -112,7 +56,20 @@ Validates that the user defined filters for a subscription:
 - values are coercable to the correct column type
 */
 declare
-    col_names text[] = cdc.selectable_columns(new.entity);
+    col_names text[] = coalesce(
+            array_agg(c.column_name order by c.ordinal_position),
+            '{}'::text[]
+        )
+        from information_schema.columns c
```
Comment (soedirgo) [inline sql/walrus--0.1.sql:63]: Curious, does it have any effect when using `pg_attribute` instead?

File: sql/walrus--0.1.sql [hunk 2]
```diff
@@ -112,7 +56,20 @@ Validates that the user defined filters for a subscription:
...
+        where
+            (c.table_schema || '.' || c.table_name)::regclass = new.entity
```
Comment (soedirgo) [inline sql/walrus--0.1.sql:65]: Should this be quoted with `quote_ident`?

**Review by soedirgo — APPROVED** [review]

File: sql/walrus--0.1.sql (hunk 2 above)
Comment (olirice) [reply sql/walrus--0.1.sql]: Yes it should, thanks!

File: sql/walrus--0.1.sql (hunk 1 above)
Comment (olirice) [reply sql/walrus--0.1.sql:63]: the pg_catalog tables do tend to respond faster than information_schema (haven't looked into why) but to get the same information there would need to be extra joins to filter the namespace by name vs oid

There's also a weird edge case with casting to regclass where, if the schema is in `search_path` it gets dropped from the regclass text representation 

e.g.
```sql
'public.my_table'::regclass => 'my_table`
```
which then makes it difficult to extract the schema reliably for filtering without an extra query

**Pull request merged**